Improve visual styling of team page

manuelmeister opened the pull request:
This PR tries to improve the workflow of the team page. It is currently not yet functional.

|before|after|
|-|-|
| <img width="876" alt="Bildschirmfoto 2023-03-14 um 17 17 46" src="https://user-images.githubusercontent.com/3001985/224969911-2dae3279-9ad6-43a4-852c-4e9494b425fe.png"> | ![Bildschirmfoto 2023-05-26 um 19 23 39](https://github.com/ecamp/ecamp3/assets/3001985/45f8cb3f-29f8-4bd6-b144-db2e3d46c122)  |
| <img width="375" alt="Bildschirmfoto 2023-03-14 um 17 18 14" src="https://user-images.githubusercontent.com/3001985/224969957-63ff5069-fe3d-4a19-a563-71bf30a96da4.png"> |![Bildschirmfoto 2023-05-26 um 19 24 18](https://github.com/ecamp/ecamp3/assets/3001985/23bb4ea2-9c1b-4236-9119-42e93e186ef9)  |

![Bildschirmfoto 2023-05-26 um 19 25 03](https://github.com/ecamp/ecamp3/assets/3001985/a4859d42-093d-4385-a588-656e699df3e4)
![Bildschirmfoto 2023-05-26 um 19 25 37](https://github.com/ecamp/ecamp3/assets/3001985/19b60adc-1c47-4bd3-9c04-c1435fd49965)

manuelmeister committed:
Draft: improve visual styling of team page

carlobeltrame commented on frontend/src/components/collaborator/CollaboratorListItem.vue:
```diff
@@ -1,104 +1,111 @@
-        {{ collaborator.inviteEmail }}
+      <v-list-item-subtitle>
+        {{ $tc(roles[collaborator.role].translation) }} &middot;
+        <template v-for="icon in roles[collaborator.role].icons">
+          <v-icon :key="icon" x-small>{{ icon }}</v-icon
+          >&nbsp;
+        </template>
```
These icons represent the permissions which this person has in the camp, right? Are they really understandable this way? I had to think about it for a while until I got it.
Initially I thought these are clickable or mean separate things (as in, can be active or inactive independently of each other). Maybe you could reduce the spacing of the icons a little, so that it is clearer that they belong together..?
Also, the icons are stuck to the text baseline. I feel like it would be nice if they were vertically centered with the text (especially the middot) instead. EDIT: I noticed that only some of the icons are too low on my screen:
![Screenshot 2023-03-14 at 14-00-34 eCamp3](https://user-images.githubusercontent.com/7566995/225008810-1bcee9de-53db-4f04-9115-60391279b62a.png)
The active collaborator items are clearly too low, while the invitations icon is better (could be another pixel higher, but is also okay the way it is). Is there any difference in styling, or is this some font hinting issue?

manuelmeister replied:
Yes. I will fix this issue. Thanks for pointing out!

Sadly I can't test on non retina devices., I will try to look at the centering.

carlobeltrame replied:
You can use our browserstack account. I think I have invited you all previously, but let me know if you need another invite

carlobeltrame replied:
Or at least in Firefox, in responsive design mode, you can simulate DPRs other than your device:
![Screenshot from 2023-03-15 11-17-11](https://user-images.githubusercontent.com/7566995/225279834-4cd4b4b4-3808-4143-ae6f-bb763a5f1f2e.png)
But not sure how complete this simulation is...

manuelmeister replied:
I would not rely on that. Also reviewing non retina content on a retina screen is not advised.

carlobeltrame replied:
Okay, but browserstack should still work, right. There, the website is rendered on real devices in real browsers which aren't dependent on your resolution, and this is then streamed to your retina screen.

manuelmeister replied:
@carlobeltrame I'm not sure If this is resolved. Can you please check again?

carlobeltrame replied:
![Screenshot 2023-05-29 at 16-28-11 eCamp3](https://github.com/ecamp/ecamp3/assets/7566995/e625c7bf-ae80-4c37-ba15-ca1e0bb9e579)


The issue is still present. I noticed that if I remove the `vertical-align: middle` rule from the `.v-icon.v-icon` class, it looks as expected. (But I think Vuetify creates this rule and it's important for the application logo, so maybe override it for these icons?)

manuelmeister replied:
Should be solved now

carlobeltrame replied:
Still not, no. Your new rule `.vertical-baseline` has lower specificity than the vuetify rule `.v-icon.v-icon`. Sorry to keep you up with this very slight detail, but you are really not in the dark here. Browserstack is intended exactly for this. Don't be afraid to use it :)

carlobeltrame replied:
Thanks, now it works :) will review the rest soon

manuelmeister replied:
Sorry, this mistake has slipped through. Now I've tested it on Mac Chrome and Windows Firefox and it is both correctly aligned. But I could not find a Desktop Linux on Browserstack.

carlobeltrame commented on frontend/src/views/camp/Collaborators.vue:
```diff
@@ -39,67 +39,27 @@ Displays collaborators of a single camp.
+      <div class="text-right">
+        <dialog-collaborator-create v-if="isManager" :camp="camp()">
+          <template #activator="{ on }">
+            <button-add color="success" icon="mdi-account-plus" v-on="on">
```
Have you thought about moving this button up, either to the active collaborators section (mental model "Add a new collaborator") or to the invitations section (mental model "send out a new invitation")?

I feel like this button is one of the most often used features, way more often than the inactive collaborators list.

manuelmeister replied:
I tried putting it into the toolbar, but it seemed off. I will try to add a button on the left hand side under Mitarbeitende. (because this will always be there, as you have to be a manager).

manuelmeister replied:
Moved it to the Members and handled it just like on camp admin

carlobeltrame reviewed (commented):
Visually looks great to me :) Haven't looked at the code, obviously.

Have you thought about ordering the collaborators by role?

When removing a collaborator (setting to inactive as well as completely deleting from the camp), would some info on how many responsibilities, material items etc. be useful to decide whether anything needs to be cleaned up before removing the person?

manuelmeister commented:
Nice idea with the responsibilities and material lists.
I think ordering would make sense.

usu commented:
Visually very nice. Also haven't looked at the code, yet. A few comments from clicking through:

- Deactivating an active user opens to dialog/modals on top of each other. As far as I remember, we have tried to avoid this so far. Let's discuss, if we want to deviate from this.
- When editing the role, the dropdown is not populated with the current value.
- When editing the role, cannot save it --> The Role field is required.
- Inviting again --> would be nice to add a spinner to the button (problem can easily be seen when enabling slow 3G throttling)
- Deactivate --> same: add a spinner
- When inviting a new user --> cannot submit --> The role field is required
- Permission icons --> should we add a tooltip which explains the permission rights?
- Resend invite: Maybe a toast would be nice after resending is successful. I think people could easily miss that something happened and then clicking on the button repeatedly.

<img width="311" alt="image" src="https://user-images.githubusercontent.com/449555/225104642-759a50d7-7641-4862-a5f2-1c1ba7973595.png">

<img width="304" alt="image" src="https://user-images.githubusercontent.com/449555/225104894-96e52378-c660-45a9-9e28-e996aacf3c7b.png">

<img width="304" alt="image" src="https://user-images.githubusercontent.com/449555/225104957-d2a8c9b9-459d-4647-9991-7d9cf79694f4.png">

<img width="303" alt="image" src="https://user-images.githubusercontent.com/449555/225105017-2012925b-89c9-45ea-8f44-ab948db4c870.png">

manuelmeister committed:
Add tooltip and improve visual hierarchy of abilities icon hint

manuelmeister committed:
Improve role select items

manuelmeister commented:
@usu should we close the other one while in deleting dialog? Because I'm not sure how to reduce complexity otherwise. Because on Mobile it looks like a different view. Maybe we add delete buttons on desktop?

BacLuc commented:
### "Core" Meeting Discussion

@carlobeltrame will think about how to implement this, and then discuss with @manuelmeister on how to finish this feature.

manuelmeister commented:
## Discussion with @carlobeltrame 

To prevent double dialogs on desktop screen, I'll change the edit dialog on desktop to a bottom sheet. If this design proves successful, I'll implement it for the camp admin as well.

To resolve the icon alignment we set `inline-flex` on the button.

manuelmeister committed:
Merge remote-tracking branch 'upstream/devel' into improve/team-ux

# Conflicts:
#	frontend/src/views/camp/Collaborators.vue

manuelmeister committed:
Merge remote-tracking branch 'upstream/devel' into improve/team-ux

manuelmeister committed:
Remove diabolic colon

manuelmeister commented:
@usu OMG, I wasted more than two hours to fix the problem you described. It was just a superfluous colon in front of `v-model` attribute. 🤦🏼‍♂️

manuelmeister commented on frontend/src/components/collaborator/DialogCollaboratorForm.vue:
```diff
@@ -0,0 +1,64 @@
+<template>
+  <e-select
+    :v-model="localCollaboration.role"
```
⚠️

manuelmeister committed:
Merge remote-tracking branch 'upstream/devel' into improve/team-ux

manuelmeister committed:
Rework team (camp collaborators) page

manuelmeister committed:
Optimize overall a11y

manuelmeister commented:
Instead of a dialog, I want to use a bottom sheet for changes. This will also apply to the settings page, where I would do the same as shown here. I think the "double dialog" is now not as bad as it was before.

manuelmeister committed:
Move invitation button to the top, just like on the admin page

BacLuc commented on frontend/src/components/buttons/ButtonAdd.vue:
```diff
@@ -6,10 +6,10 @@
     v-bind="$attrs"
     v-on="$listeners"
   >
+    <v-icon :left="!hideLabel" size="150%">{{ icon }}</v-icon>
```
Is this better if the icon is in front?

manuelmeister replied:
Yes, I want to unify the whole UI.

BacLuc commented on frontend/src/components/buttons/ButtonDelete.vue:
```diff
@@ -1,15 +1,20 @@
+    <v-icon v-else-if="!text" :left="$vuetify.breakpoint.smAndUp" size="150%">{{
+      icon
+    }}</v-icon>
+    <v-icon v-if="!iconOnly && text" :left="$vuetify.breakpoint.smAndUp" size="150%">{{
```
do we still need the iconOnly flag?

manuelmeister replied:
I think so, why?

BacLuc commented on frontend/src/components/buttons/ButtonDelete.vue:
```diff
@@ -32,4 +37,8 @@ export default {
-<style scoped></style>
+<style scoped>
+.ec-btn-delete:hover {
+  color: #f44336 !important;
```
braucht es das !important?

manuelmeister replied:
Ja, weil Vuetify auch !important verwendet :(

BacLuc commented on frontend/src/components/buttons/IconButton.vue:
```diff
@@ -6,10 +6,18 @@
     </span>
-    <v-icon :right="!hideLabel" size="150%" :class="{ animate: animate }">
+    <v-icon
+      v-if="!iconFirst"
```
do we still need the !iconFirst case?

manuelmeister replied:
I think I would still keep it. If for whatever reason it needs to be used, it would be unnecessary to do it again.

manuelmeister replied:
I changed it to iconLast. So it can still be used in an edgecase

BacLuc commented on frontend/src/components/collaborator/CollaboratorListItem.vue:
```diff
@@ -1,104 +1,77 @@
-        }}</span>
-      </v-tooltip>
+    <v-list-item-action class="e-collaborator-item__actions ml-2">
+      <button-edit icon-first color="primary--text" text class="my-n1 v-btn--has-bg" />
```
Don't you need to hide this for guests?
Did i get it correctly that we should not use disabled buttons, but hide the buttons if the user does not have the permission to perform an action?

carlobeltrame replied:
From an accessibility standpoint, using html5 `disabled="disabled"` is the same as hiding.

If you want the user to see the button which they aren't allowed to use, you should show a normal button and use JS `@click` to display an explanation instead of performing the action.

manuelmeister replied:
Now I do not show the button at all if the user has no rights to edit.

BacLuc commented on frontend/src/components/collaborator/CollaboratorEdit.vue:
```diff
@@ -0,0 +1,199 @@
+    },
+    reinvite() {
+      this.resendingEmail = true
+      this.api.patch(this.collaborator, { status: 'invited' }).then(() => {
```
```suggestion
      this.api.patch(this.collaborator, { status: 'invited' }).finally(() => {
```
Else the animation does not stop when an error occurs

BacLuc commented on frontend/src/components/dialog/SettingsForm.vue:
```diff
@@ -0,0 +1,171 @@
+  name: 'SettingsForm',
+  components: { ValidationObserver, ServerError },
+  props: {
+    value: { type: Boolean, required: true },
+
+    icon: { type: String, default: '', required: false },
+    title: { type: String, default: '', required: false },
+
+    submitAction: { type: Function, default: null, require: false },
+    submitIcon: { type: String, default: 'mdi-send-variant', required: false },
+    submitLabel: {
+      type: String,
+      default: function () {
+        return this.$tc('global.button.submit')
+      },
+      required: false,
+    },
+    submitColor: { type: String, default: 'primary', required: false },
+    submitEnabled: { type: Boolean, default: true, required: false },
+
+    cancelAction: { type: Function, default: null, required: false },
+    cancelIcon: { type: String, default: 'mdi-window-close', required: false },
+    cancelLabel: {
+      type: String,
+      default: function () {
+        return this.$tc('global.button.cancel')
+      },
+      required: false,
+    },
+    cancelColor: { type: String, default: 'secondary', required: false },
+    cancelEnabled: { type: Boolean, default: true, required: false },
+    cancelVisible: { type: Boolean, default: true, required: false },
+
+    loading: { type: Boolean, default: false, required: false },
+    error: { type: [Object, String, Error], default: null, required: false },
+
+    maxWidth: { type: String, default: '600px', required: false },
```
i have the feeling that this component does too much.

manuelmeister replied:
This is the same as DialogForm. But I think this could be simplified. I did not want to remove anything before we agree on this.

BacLuc commented on common/locales/de.json:
```diff
@@ -92,9 +92,12 @@
     "camp": {
       "collaborators": {
         "guest": "Gast",
+        "guestAbilities": "Nur Leserecht",
```
do we need this in the common locales?

BacLuc commented on common/locales/de.json:
```diff
@@ -92,9 +92,12 @@
-        "manager": "Verwalter",
-        "member": "Mitglied"
+        "manager": "Administration",
+        "managerAbilities": "Vollständige Lageradministration, sowie Lese- & Schreibrecht",
```
do we need this in the common locales?

manuelmeister replied:
I think it is useful, as it is not a sentence and more a state, that can be used in E-mails as well.

BacLuc replied:
But for emails we currently can only use the translations in the backend

manuelmeister replied:
Okay. Would you prefer these keys in the frontend locales?

BacLuc commented on common/locales/de.json:
```diff
@@ -92,9 +92,12 @@
+        "manager": "Administration",
+        "managerAbilities": "Vollständige Lageradministration, sowie Lese- & Schreibrecht",
+        "member": "Mitglied",
+        "memberAbilities": "Lese- & Schreibrecht"
```
do we need this in the common locales?

BacLuc commented on frontend/src/components/collaborator/CollaboratorEdit.vue:
```diff
@@ -0,0 +1,199 @@
+<template>
+  <SettingsForm
```
which functionality of the SettingsForm are you using in this component?

manuelmeister replied:
Now renamed as DetailPane will be used for all kinds of details views, that are not own pages.

BacLuc commented on frontend/src/components/dialog/SettingsForm.vue:
```diff
@@ -0,0 +1,171 @@
+<template>
+  <v-bottom-sheet
```
does this move the sheet on the bottom?
On my 4k monitor i had to search a little before i found the form.

![grafik](https://github.com/ecamp/ecamp3/assets/1506818/40228dab-84d0-4683-aa84-9e471019c5da)

manuelmeister replied:
Changed design to dialog

BacLuc reviewed (changes requested):
It looks very good.

I am note sure whether putting the form on the bottom of the page works for all screen sizes.

carlobeltrame commented on common/locales/de.json:
```diff
@@ -117,7 +120,16 @@
+      },
+      "status": {
+        "established": "Aktiv",
+        "inactive": "Deaktiviert",
```
```suggestion
        "inactive": "Inaktiv",
```

manuelmeister committed:
Fix wording of inactive in german

manuelmeister committed:
Fixed alignment of icons and removed unnecessary chunk

manuelmeister committed:
Refactor team ui

Use a dialog on desktop and a popover prompt for the deletion & deactivation
Simplified whole dialog components with DialogUiBase

manuelmeister committed:
Move from iconFirst to iconLast

manuelmeister committed:
Fix import

manuelmeister committed:
Fix translations

manuelmeister committed:
Better icon for warning prompt

manuelmeister commented:
New Popover
<img width="375" alt="Bildschirmfoto 2023-05-30 um 23 16 39" src="https://github.com/ecamp/ecamp3/assets/3001985/dd07b138-fb68-4236-859c-67dc0e99f4f2">
<img width="1000" alt="Bildschirmfoto 2023-05-30 um 23 14 06" src="https://github.com/ecamp/ecamp3/assets/3001985/b9afaf07-2052-4765-8cdc-4036caa12385">
<img width="925" alt="Bildschirmfoto 2023-05-30 um 23 15 52" src="https://github.com/ecamp/ecamp3/assets/3001985/3260aaf1-93e6-430b-98cc-7594202aa2a2">

manuelmeister committed:
Move abilities translation into frontend

manuelmeister committed:
Fix button overflow & wrap

manuelmeister commented:
@carlobeltrame I think I would move advanced features such as ordering according to role and displaying how many responsible activities in a good-first-issue PR. What do you think?

carlobeltrame commented:
> @carlobeltrame I think I would move advanced features such as ordering according to role and displaying how many responsible activities in a good-first-issue PR. What do you think?

Sounds good to me :)

usu commented on frontend/src/components/collaborator/CollaboratorForm.vue:
```diff
@@ -0,0 +1,154 @@
+        {
+          key: 'manager',
+          role: $tc('entity.camp.collaborators.manager'),
+          abilities: $tc('entity.camp.collaborators.managerAbilities'),
```
This key is not translated

<img width="1013" alt="image" src="https://github.com/ecamp/ecamp3/assets/449555/d00410d3-48f8-4951-bdf7-5d09e0e1395b">

usu commented on common/locales/fr.json:
```diff
@@ -169,7 +169,18 @@
+    "datetime": {
+      "dateLong": "dd D.MM.YYYY",
+      "dateShort": "dd D.M.",
+      "dateTimeLong": "dd D.MM.YYYY HH:mm",
```
Is there a reason we use "dd L HH:mm" for all other languages but not for French? Similar for `dateLong` as well.

manuelmeister replied:
I did it because the date format for french was not correct. I'll move this to a separate PR, where I introduce a customizable date format. Because one might want to use eCamp in English, but not in "football-is-mostly-played-with-hands" format ;) 
[#3217](https://github.com/ecamp/ecamp3/issues/3217)

usu commented on frontend/src/components/buttons/ButtonEdit.vue:
```diff
@@ -1,20 +1,25 @@
-      <slot>{{ $tc('global.button.edit') }}</slot>
-    </span>
-  </v-btn>
+  <IconButton
```
Nice to use the generic IconButton. Should we do this also for Edit/Delete/Add?

manuelmeister replied:
Probably. But I rather look at the usages generally. I want to establish a concept when to use which button style, in a way that serves the primary use case and is not too distracting for secondary actions.

usu commented on frontend/src/components/buttons/ButtonDelete.vue:
```diff
@@ -1,15 +1,18 @@
   >
     <v-icon v-if="iconOnly">{{ icon }}</v-icon>
-    <v-icon v-else :left="$vuetify.breakpoint.smAndUp" size="150%">{{ icon }}</v-icon>
+    <v-icon v-else-if="!text" :left="$vuetify.breakpoint.smAndUp">{{ icon }}</v-icon>
```
General question to all the changes to the Button components: As we use these components and quite many places, I'd assume you did some sanity check that layout elsewhere is not broken, correct?

manuelmeister replied:
Yes, I could not spot a problem. Most of the UIs with buttons will be changed in my future camp admin PR anyway.

usu reviewed (commented):
Haven't looked at all the code yet. But look & feel is really nice, love it ❤️

manuelmeister committed:
Move translations into a computed

manuelmeister committed:
Revert french translations back to L format

manuelmeister committed:
Fix changed translation path

BacLuc commented on frontend/src/locales/en.json:
```diff
@@ -109,20 +109,27 @@
+        "delete": "Do you really want to remove collaborator '{name}'?",
+        "inviteAgain": "Invite again",
+        "resendEmail": "Resend invite",
+        "resentEmail": "Invitation sent",
```
i would have called this key resendEmailFinished, that it differs more from resendEmail

manuelmeister commented:
Created issues for further improvements:
- #3492
- #3493

manuelmeister committed:
Fix inline baseline alignment

carlobeltrame commented on common/locales/de.json:
```diff
@@ -117,7 +117,16 @@
     },
     "campCollaboration": {
-      "inactive": "inaktiv"
+      "fields": {
+        "inviteEmail": "E-Mail",
+        "role": "Rolle",
+        "status": "Status"
+      },
+      "status": {
+        "established": "Aktiv",
+        "inactive": "Inaktiv",
+        "invited": "Eingeladen"
+      }
```
I still think most of these aren't needed in print, and therefore should just live in the frontend translations. But it's not just these, and we can open the discussion again when we next meet.

I really like the translation keys `entity.campCollaboration.status.xyz` though :)

manuelmeister replied:
Should I open a new Meeting Discuss for how we want to handle translations. (small adjustments)

carlobeltrame commented on frontend/src/components/buttons/ButtonDelete.vue:
```diff
@@ -1,15 +1,18 @@
     <v-icon v-if="iconOnly">{{ icon }}</v-icon>
-    <v-icon v-else :left="$vuetify.breakpoint.smAndUp" size="150%">{{ icon }}</v-icon>
+    <v-icon v-else-if="!text" :left="$vuetify.breakpoint.smAndUp">{{ icon }}</v-icon>
+    <v-icon v-if="!iconOnly && text" :left="$vuetify.breakpoint.smAndUp">{{
```
If I understand correctly, you could use `v-else` here, I think that might be easier to understand?
```suggestion
    <v-icon v-else :left="$vuetify.breakpoint.smAndUp">{{
```

usu commented on frontend/src/components/generic/DetailPane.vue:
I like this approach 👍  
I guess the vision would be to roll this out to all other dialogs we use in our application, right?

I'm not 100% confident with the naming. Purely from the naming of the components, it's not directly clear what DetailPane, DialogBottomSheet and DialogForm does and what the difference is between each of them. In absence of a better proposal and to not block this PR, I'm absolutely happy if we refactor this, once we want to roll out the same approach to other dialogs.

manuelmeister replied:
For me pure BottomSheets and Dialogs are still valid. DetailPane should be used as a detail view of an item in the context of a list. In other cases a dialog may still be valid. Also i think, the BottomSheet could be used to further configure the picasso view for instance. 

I could rename them
- DetailForm
- BottomForm
- DialogForm

I will continue this in the next PR: #3498

carlobeltrame replied:
I was also confused by the naming of DetailPane, which looks and behaves just like a dialog on desktop. I like the idea to name these components after how they are used, not how they look, because they now look differently on desktop and mobile.

manuelmeister replied:
Let's continue the discussion in the following Issue #3498

usu commented on frontend/src/components/dialog/DialogForm.vue:
```diff
@@ -84,50 +84,22 @@
-    }
+  computed: {
+    currentlySaving() {
+      return this.isSaving || this.savingOverride
```
Why exactly is the `savingOverride` functionality necessary? Haven't really figured it out by looking at the code.

manuelmeister replied:
This comes from the problem that DialogForm and DialogBottomSheet are interlocked with DetailPane. If you save and then reduce viewport width, then you could submit again, if the isSaving is not synced across components.

carlobeltrame replied:
Crazy that you thought of this. Is this a real use case though? If the user really wanted to submit twice, they could also open two tabs. So we need consistency validation in the backend anyways. Based on your description, I think we could leave this savingOverride mechanism out if it makes the UI code simpler.

manuelmeister replied:
It just happened to me 🤷.
I was trying out the feature deployment on the train and resized the window and suddenly it was submittable again, which is confusing ux-wise.

carlobeltrame replied:
Yes, sure, during development. But the average user will never run into this I think, and if they do, it's no big problem if they submit a form twice.

manuelmeister replied:
Maybe in this case it is not tragic. But other cases the resulting ux may be worse.

usu commented on frontend/src/components/dialog/DialogForm.vue:
```diff
@@ -139,11 +111,14 @@ export default {
   methods: {
     async doSubmit() {
```
There is still quite some code overlap between DialogForm and DialogBottomSheet. I reckon you don't want to put this into DialogUiBase, because DialogUiBase is also used by DetailPane.

Should we put this code overlap in a separate mixin?

manuelmeister replied:
#3497

usu commented on frontend/src/components/layout/ContentGroup.vue:
```diff
@@ -5,10 +5,15 @@ Displays the content wrapped inside a card.
     <slot name="title">
-      <div v-if="title" class="ec-content-group__title py-1 subtitle-1">
+      <component
+        :is="titleTag"
```
I think you didn't use the `titleTag` property anywhere. Is this change still needed?

manuelmeister replied:
It is needed for a11y. Currently if it is used inside ContentCard the usage is correct. But the nesting could also require a h3

usu commented on frontend/src/components/layout/ContentGroup.vue:
```diff
@@ -5,10 +5,15 @@ Displays the content wrapped inside a card.
         <v-icon v-if="icon" left>{{ icon }}</v-icon>
         {{ title }}
-      </div>
+        <slot name="title-actions" />
```
We probably want to refactor other usages of `ContentGroup` with this new functionality of title-actions. Do we want to create an issue to keep track of this?

manuelmeister replied:
#3496

usu commented on frontend/src/locales/de.json:
```diff
@@ -109,20 +109,27 @@
       },
-      "collaboratorListItemDeactivate": {
+      "collaboratorForm": {
+        "roleHint": "Jedes Lager benötigt mindestens eine Person mit Administratorrechten."
```
```suggestion
        "roleHint": "Jedes Lager benötigt mindestens eine Person mit Administrationsrechten."
```

usu reviewed (approved):
A few comments and questions below. I'd also be happy to merge as-is if we properly keep track of the cleanup activities.

manuelmeister committed:
Fix german administrator role hint wording

Co-authored-by: Urban Suppiger <urban@suppiger.net>

manuelmeister commented:
I've added the leftover todos in #3498

carlobeltrame commented on frontend/src/components/collaborator/CollaboratorForm.vue:
```diff
@@ -0,0 +1,139 @@
+            aria-describedby="readonly"
+            :name="$tc('entity.campCollaboration.fields.role')"
+            :items="items"
+            :hint="$tc('components.collaborator.collaboratorForm.roleHint')"
```
I think we could remove this copy of the hint. It is shown anyways in the tooltip when it's relevant. Also, right now, it's shown also on the collaborator create form, where it does not really make sense (I mean, it's good to know, but not really relevant there).

manuelmeister replied:
#3499

carlobeltrame commented on frontend/src/components/collaborator/CollaboratorForm.vue:
```diff
@@ -0,0 +1,139 @@
+      :name="$tc('entity.campCollaboration.fields.role')"
+      fieldname="role"
+      :items="items"
+      :hint="$tc('components.collaborator.collaboratorForm.roleHint')"
```
Same here, see above

manuelmeister replied:
#3499

carlobeltrame commented on frontend/src/components/collaborator/CollaboratorForm.vue:
```diff
@@ -0,0 +1,139 @@
+        },
+      ]
+    },
+    localCollaboration() {
+      return this.collaboration
+    },
```
Couldn't you remove this? I think this just returns the reference to the prop object anyways, so you are not keeping a local copy here. So when you mutate the role via the form, you mutate `this.collaboration` directly.
So it makes no difference to keep a computed here, and the name evokes the wrong impression of a local copy.

manuelmeister replied:
We use it this way already. This is a workaround for eslint which would otherwise complain. This would need to be reworked fully. I'm not sure how to handle this sensible. #3500

carlobeltrame commented on frontend/src/components/collaborator/CollaboratorListItem.vue:
```diff
@@ -1,138 +1,88 @@
-          this.resendingEmail = false
-        })
+    name() {
+      return this.collaborator.user
+        ? this.collaborator.user().displayName
+        : this.collaborator.inviteEmail
```
We have helpers for this, e.g. `campCollaborationDisplayName(this.collaborator, this.$tc.bind(this), false)` would do the same and is unit tested.

manuelmeister replied:
#3501

manuelmeister replied:
#3501

carlobeltrame commented on frontend/src/components/collaborator/CollaboratorListItem.vue:
```diff
@@ -1,138 +1,88 @@
   data: () => ({
-    resendingEmail: false,
+    roles: {
+      manager: {
+        roleTranslation: 'entity.camp.collaborators.manager',
+        abilitiesTranslation: 'global.collaborationAbilities.manager',
+        icons: ['mdi-eye-outline', 'mdi-pencil-outline', 'mdi-cog-outline'],
+      },
+      member: {
+        roleTranslation: 'entity.camp.collaborators.member',
+        abilitiesTranslation: 'global.collaborationAbilities.member',
+        icons: ['mdi-eye-outline', 'mdi-pencil-outline'],
+      },
+      guest: {
+        roleTranslation: 'entity.camp.collaborators.guest',
+        abilitiesTranslation: 'global.collaborationAbilities.guest',
+        icons: ['mdi-eye-outline'],
+      },
```
Consider extracting these roles into a separate file and reusing them for the dropdown select

manuelmeister replied:
Where would you place such a file?

carlobeltrame commented on frontend/src/components/prompt/PopoverPrompt.vue:
```diff
@@ -0,0 +1,184 @@
+<template>
+  <v-menu
```
I really love the way this looks. You outdid yourself with this one :) I also tried to break it on ridiculously tiny screen sizes, but it was always useable.

carlobeltrame commented on frontend/src/components/prompt/PromptEntityDelete.vue:
```diff
@@ -0,0 +1,46 @@
+    <template #activator="scope">
+      <slot name="activator" v-bind="scope" />
+    </template>
+    <slot>{{ $tc('components.prompt.promptEntityDelete.warningText') }}</slot>
```
We could even introduce a "type the name of the entity into this text field before deleting" feature in this component in the future :+1:

carlobeltrame reviewed (approved):
I'm fine with merging now, cleanup later as well if you'd like to wrap this up. Let's just make sure to track all cleanup tasks properly.
